main.ts

```diff
@@ -1,135 +1,110 @@
-import { MarkdownView, Plugin, EditorPosition, Editor } from "obsidian";
-
+import { COMMANDS } from "const";
+import { Editor, EditorPosition, Plugin } from "obsidian";
+
+// Override the obsidian module by adding the non-documented Editor methods
+// This makes it so that TypeScript stops shouting at you for accessing a property that it thinks doesn't exist
+// And gives you type-safety
+declare module "obsidian" {
+	interface Editor {
+		cm: {
+			state?: { wordAt: (offset: number) => EditorSelection };
+			findWordAt?: (pos: EditorPosition) => EditorSelection;
+		};
+	}
+}
 export default class SmarterMDhotkeys extends Plugin {
 	async onload() {
-		this.addCommand({
-			id: "smarter-asterisk-bold",
-			name: "Smarter Bold",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("**", "**", editor),
-		});
-		this.addCommand({
-			id: "smarter-underscore-bold",
-			name: "Smarter Underscore Bold",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("__", "__", editor),
-		});
-		this.addCommand({
-			id: "smarter-asterisk-italics",
-			name: "Smarter Italics",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("*", "*", editor),
-		});
-		this.addCommand({
-			id: "smarter-underscore-italics",
-			name: "Smarter Underscore Italics",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("_", "_", editor),
-		});
-		this.addCommand({
-			id: "smarter-comments",
-			name: "Smarter Comments",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("%%", "%%", editor),
-		});
-		this.addCommand({
-			id: "smarter-inline-code",
-			name: "Smarter Inline Code",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("`", "`", editor),
-		});
-		this.addCommand({
-			id: "smarter-highlight",
-			name: "Smarter Highlight",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("==", "==", editor),
-		});
-		this.addCommand({
-			id: "smarter-strikethrough",
-			name: "Smarter Strikethrough",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("~~", "~~", editor),
-		});
-		this.addCommand({
-			id: "smarter-wikilink",
-			name: "Smarter wikilink (internal link)",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("[[", "]]", editor),
-		});
-		this.addCommand({
-			id: "smarter-md-link",
-			name: "Smarter Markdown Link",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("[", "]()", editor),
+		COMMANDS.forEach((command) => {
+			const { id, name, before, after } = command;

```

Comment:
lol, I didn't even know you could do that. is that a typescript thing or js?

Reply:
Which part?

Reply:
Whenever I ask this question myself, the answer is usually: "If it has anything to do with types, then it's TS, everything else is JS"

Reply:
declaring all those variables at once. Guess this is Object destructuring then, analogue to array destructuring?

Reply:
Ahh right, yes.

`const thing = obj.thing` is the same as `const { thing } = obj`. You _destructure_ the property off the obj.

```js
const a = 1
const b = 2
```
is the same as:
```js
const [a, b] = [1, 2]
```